enclave.go

```diff
@@ -67,14 +67,49 @@ type Enclave struct {
 
 // Config represents the configuration of our enclave service.
 type Config struct {
+	// SOCKSProxy must be set if
+	//   1) your enclave application should obtain a Let's Encrypt-signed
+	//      certificate (i.e., if UseACME is set to true)
+	// or if
+	//   2) your enclave application makes HTTP requests over the Internet.
+	// If so, set SOCKSProxy to "socks5://127.0.0.1:1080".
 	SOCKSProxy string
-	FQDN       string
-	Port       int
-	UseACME    bool
-	Debug      bool
-	FdCur      uint64
-	FdMax      uint64
-	AppURL     string
+
+	// FQDN contains the fully qualified domain name that's set in the HTTPS
+	// certificate of the enclave's Web server, e.g. "example.com".
+	FQDN string
+
+	// Port contains the TCP port that the Web server should listen on, e.g.
+	// 8443.  Note that the Web server listens for this port on the private
+	// VSOCK interface.  This is not an Internet-facing port.
+	Port int
+
+	// UseACME must be set to true if you want your enclave application to
+	// request a Let's Encrypt-signed certificate.  If this is set to false,
+	// the enclave creates a self-signed certificate.
+	UseACME bool
+
+	// Debug can be set to true to see debug messages, i.e., if you are
+	// starting the enclave in debug mode by running:
```

Comment:
It would be nice if the framework could ask the supervisor if it's in debug mode, but I don't see an interface for that.

Reply:
One could implement this by asking for an attestation document and checking if its PCR values are all zeroed out. I filed https://github.com/brave/nitriding/issues/25 for this.

Reply:
Ah, clever!